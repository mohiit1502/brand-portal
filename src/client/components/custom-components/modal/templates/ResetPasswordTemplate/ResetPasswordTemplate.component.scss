$theme-color: #017dc6;

.c-ResetPasswordTemplate {

  .modal-dialog{
    .modal-content{
      border-bottom: 9px solid #7ac4e4;
      .modal-header{
        background: #006799;
        color: white;
        font-size: 27px;
        font-weight: bold;

      }
      .modal-body{
        //border-bottom: ;
        .c-ButtonsPanel{
          .reset-password-button-panel{
            .btn-primary {
              background: #006799;
              font-weight: bold;
              border-radius: 42px;
              border: none;
              font-size: 15px;
            }

            .btn-primary:disabled {
              background: #a3a8a9;
            }
            .cancel-btn{
              font-weight: bold;
              text-decoration: underline;
              font-size: 16px;
            }
          }
        }
      }
    }
  }

  .icon-view-password, .icon-hide-password {
    cursor: pointer;
    display: block;
    content: "";
    position: absolute;
    top: 0.3rem;
    right: 0;
    width: 40px;
    height: 25px;
    background-repeat: no-repeat;
    z-index: 10;
  }

  .icon-view-password {
    background-image: url("./../../../../../images/showPassword.svg");
  }

  .icon-hide-password {
    background-image: url("./../../../../../images/hidePassword.svg");
  }
}
