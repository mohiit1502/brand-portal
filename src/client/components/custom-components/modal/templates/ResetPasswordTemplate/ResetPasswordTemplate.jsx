import React, {Component} from 'react';
import PropTypes from "prop-types";
import {connect} from "react-redux";
import Helper from "../../../../../utility/helper";
import Http from "../../../../../utility/Http";
import ContentRenderer from "../../../../../utility/ContentRenderer";
import {NOTIFICATION_TYPE, showNotification} from "../../../../../actions/notification/notification-actions";
import {TOGGLE_ACTIONS, toggleModal} from "../../../../../actions/modal-actions";
import CONSTANTS from "../../../../../constants/constants";
import './ResetPasswordTemplate.component.scss';
import mixpanel from "../../../../../utility/mixpanelutils";
import MIXPANEL_CONSTANTS from "../../../../../constants/MixPanelConsants";

class ResetPasswordTemplate extends Component {
  constructor(props) {
    super(props);
    const functions = ["bubbleValue", "onChange", "resetTemplateStatus", "handleSubmit", "validateForm"];
    functions.forEach(name => this[name] = this[name].bind(this));
    this.loader = Helper.loader.bind(this);
    this.getFieldRenders = ContentRenderer.getFieldRenders.bind(this);
    const resetPasswordConfiguration = this.props.resetPasswordConfiguration ? this.props.resetPasswordConfiguration : {}
    this.state = {
      section: {...resetPasswordConfiguration.sectionConfig},
      form: {
        inputData: resetPasswordConfiguration.fields,
        ...resetPasswordConfiguration.formConfig,
      }
    };
  }

  bubbleValue (evt, key, error) {
    const targetVal = evt.target.value;
    this.setState(state => {
      state = {...state};
      state.form.inputData[key].value = targetVal;
      state.form.inputData[key].error = error;
      return state;
    });
  }

  onChange(evt, key) {
    if (evt && evt.target) {
      const targetVal = evt.target.value;
      this.setState(state => {
        state = {...state};
        if (key === "currentPassword" && state.form.error === state.form.incorrectPasswordError) {
          state.form.error = false;
        }
        if (key === "newPassword" || key === "confirmNewPassword") {
          (state.form.error === state.form.old5PasswordsError || state.form.error === state.form.passwordPolicyMessage) && (state.form.error = false);
          state.form.inputData.errorSub.error = state.form.passwordsDifferent ? state.form.passwordMismatchError : "";
        }
        state.form.inputData[key].value = targetVal;
        state.form.inputData[key].isDirty = true;
        return state;
      }, this.checkToEnableSubmit);
    }
  }

  validateForm () {
    const form = {...this.state.form};
    const inputData = {...form.inputData};
    let hasError = false;
    form.inputData = inputData;
    Object.keys(inputData).forEach(inputKey => {
      const inputObj = inputData[inputKey];
      if (inputObj.type === "text" || inputObj.type === "password" && !inputObj.value && inputObj.required) {
        hasError = true;
        inputObj.error = inputObj.validators && inputObj.validators.validateRequired && inputObj.validators.validateRequired.error;
      }
    })
    hasError = hasError || form.passwordsDifferent || form.error;
    if (hasError) {
      this.props.showNotification(NOTIFICATION_TYPE.ERROR, form.toastMessageExistingErrors);
    }
    hasError && this.setState({form});
    return hasError;
  }

  async handleSubmit(evt) {
    evt.preventDefault();
    const form = this.state.form;
    if (!this.validateForm()) {
      const currentPassword = form.inputData.currentPassword.value;
      const newPassword = form.inputData.newPassword.value;
      const payload = {oldPassword: currentPassword, newPassword};
      const url = form.apiPath;

      this.loader("form", true);
      return Http.post(url, payload, null, null, this.props.showNotification, this.state.form.passwordChangedMessage, this.state.form.failureMessage)
        .then(res => {
          this.resetTemplateStatus();
          this.props.toggleModal(TOGGLE_ACTIONS.HIDE);
          this.loader("form", false);
          mixpanel.trackEvent(MIXPANEL_CONSTANTS.USER_PROFILE.CHANGE_PASSWORD.SAVE_PASSWORD_SUCCESS);
        })
        .catch(err => {
          this.loader("form", false);
          if (err.status === CONSTANTS.STATUS_CODE_400) {
            if (err.message && err.message.toLowerCase() === CONSTANTS.ERRORMESSAGES.PASSWORDMISMATCH.toLowerCase()) {
              this.setState(state => {
                state = {...state};
                state.form.error = state.form.incorrectPasswordError;
                return true;
              });
            } else if (err.message && err.message.toLowerCase() === CONSTANTS.ERRORMESSAGES.SAMEPASSWORD.toLowerCase() && !form.passwordsDifferent) {
              this.setState(state => {
                state = {...state};
                state.form.error = form.old5PasswordsError;
                return true;
              });
            } else if (err.message && err.message.toLowerCase() === CONSTANTS.ERRORMESSAGES.PASSWORDPOLICYMESSAGE.toLowerCase()) {
              this.setState(state => {
                state = {...state};
                state.form.error = form.passwordPolicyMessage;
                return true;
              });
            }
          }
          console.log(err);
          mixpanel.trackEvent(MIXPANEL_CONSTANTS.USER_PROFILE.CHANGE_PASSWORD.SAVE_PASSWORD_FAILURE, err);
        });
    }
  }

  resetTemplateStatus() {
    const form = {...this.state.form};
    form.inputData.currentPassword.value = "";
    form.inputData.newPassword.value = "";
    form.inputData.confirmNewPassword.value = "";

    form.inputData.currentPassword.isDirty = false;
    form.inputData.newPassword.isDirty = false;
    form.inputData.confirmNewPassword.isDirty = false;

    form.inputData.currentPassword.error = "";
    form.inputData.newPassword.error = "";
    form.inputData.confirmNewPassword.error = "";

    form.inputData.errorSub.error = "";

    this.setState({form});
    this.props.toggleModal(TOGGLE_ACTIONS.HIDE);
    mixpanel.trackEvent(MIXPANEL_CONSTANTS.USER_PROFILE.CHANGE_PASSWORD.CANCLE_CHANGE_PASSWORD);
  }

  render() {
    const section = this.state.section;
    const form = this.state.form;
    return (
      <div className="c-ResetPasswordTemplate modal fade show" id="singletonModal" tabIndex="-1" role="dialog">
        <div className="modal-dialog modal-dialog-centered" role="document">
          <div className="modal-content">
            <div className="modal-header align-items-center">
              {section.sectionTitle}
              <button type="button" className="close text-white" aria-label="Close" onClick={this.resetTemplateStatus}>
                <span aria-hidden="true">&times;</span>
              </button>
            </div>
            <div className={`modal-body text-left${this.state.form.loader ? " loader" : ""}`}>
              <form onSubmit={this.handleSubmit} className="h-100 px-2">
                <p>{form.formHeading}</p>
                {form.passwordGuidance && <small className={`form-text font-size-12 custom-input-help-text mb-3`} style={{color: "#777"}}>{ form.passwordGuidance }</small>}
                {form.error && <small className={`form-text custom-input-help-text text-danger mb-3 pl-3`}>{ form.error }</small>}
                {this.getFieldRenders()}
              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
};


ResetPasswordTemplate.propTypes = {
  modal: PropTypes.object,
  toggleModal: PropTypes.func,
  data: PropTypes.object,
  showNotification: PropTypes.func
};

const mapStateToProps = state => {
  return {
    resetPasswordConfiguration: state.content && state.content.metadata && state.content.metadata.SECTIONSCONFIG && state.content.metadata.SECTIONSCONFIG.RESETPASSWORD,
    modal: state.modal
  };
};

const mapDispatchToProps = {
  toggleModal,
  showNotification
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ResetPasswordTemplate);
