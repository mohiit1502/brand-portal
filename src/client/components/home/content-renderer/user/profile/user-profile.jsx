import React from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import Http from "../../../../../utility/Http";
import {updateUserProfile} from "../../../../../actions/user/user-actions";
import {dispatchDiscardChanges, TOGGLE_ACTIONS, toggleModal} from "../../../../../actions/modal-actions"
import {showNotification} from "../../../../../actions/notification/notification-actions"
import InputFormatter from "../../../../../utility/phoneOps";
import Helper from "../../../../../utility/helper";
import ContentRenderer from "../../../../../utility/ContentRenderer";
import Validator from "../../../../../utility/validationUtil";
// import FORMFIELDCONFIG from "./../../../../../config/formsConfig/form-field-meta";
import "../../../../../styles/home/content-renderer/user/profile/user-profile.scss";
import CONSTANTS from "../../../../../constants/constants";
import mixpanel from "../../../../../utility/mixpanelutils";
import MIXPANEL_CONSTANTS from "../../../../../constants/mixpanelConstants";
import Cookies from "electrode-cookies";

class UserProfile extends React.Component {

  constructor (props) {
    super(props);
    const functions = ["bubbleValue", "displayChangePassword", "displayManageProfileNotification", "isDirty", "onChange", "setFormData", "disableInput", "saveUser"];
    functions.forEach(name => this[name] = this[name].bind(this));
    this.validateState = Validator.validateState.bind(this);
    this.getFieldRenders = ContentRenderer.getFieldRenders.bind(this);
    this.loader = Helper.loader.bind(this);
    this.onInvalid = Validator.onInvalid.bind(this);
    this.invalid = {firstName: false, lastName: false, companyName: false, phone: false};
    const userProfileConfiguration = this.props.userProfileContent ? this.props.userProfileContent : {}

    this.state = {
      section: {...userProfileConfiguration.sectionConfig},
      form: {
        ...userProfileConfiguration.formConfig,
        inputData: {...userProfileConfiguration.fields},
        // underwritingChecked: false,
      },
      isSeller: Cookies.get("client_type") === "seller"
    };

    const formatter = new InputFormatter();
    const handlers = formatter.on(`#${this.state.section.id}-${this.state.form.inputData.phone.inputId}-custom-input`);
    this.prebounceChangeHandler = handlers.inputHandler;

    Object.keys(this.state.form.inputData).forEach(itemKey => {
      const item = this.state.form.inputData[itemKey];
      this.state.form.inputData[itemKey].value = Helper.search(item.initValuePath, this.props.userProfile);
      if  (itemKey === "phone") {
        const value = this.state.form.inputData[itemKey].value;
        this.state.form.inputData[itemKey].value = (value === "0000000000") || (value === "(000) 000-0000") ? "" : value;
      }
    });
  }

  componentDidUpdate(prevProps) {
    if (prevProps !== this.props) {
      this.setFormData(this.props.userProfile);
    }
  }

  isDirty () {
    const {firstName: originalFName, lastName: originalLName, email: originalEmail, phoneNumber: originalPhone, companyName: originalCompany} = this.props.userProfile;
    const {firstName: {value: currentFName}, lastName: {value: currentLName}, emailId: {value: currentEmail}, phone: {value: currentPhone}} = this.state.form.inputData;
    const currentCompany = this.state.form.inputData.company ? this.state.form.inputData.company.value : "";
    const originalPhoneModified = (originalPhone === "0000000000" || originalPhone === "(000) 000-0000") ? "" : originalPhone;
    return originalFName !== currentFName || originalLName !== currentLName || originalEmail !== currentEmail || originalPhoneModified !== currentPhone || (originalCompany && currentCompany && originalCompany !== currentCompany);
      // || ((!originalCompany && currentCompany) || (originalCompany && !currentCompany) || (originalCompany !== currentCompany));
  }

  setFormData(obj) {
    if (this.props.shouldDiscard) {
      const form = {...this.state.form};
      form.isDisabled = true;
      form.inputData.firstName.value = obj.firstName;
      form.inputData.lastName.value = obj.lastName;
      form.inputData.companyName.value = obj.type === "ThirdParty" ? obj.companyName : "";
      form.inputData.emailId.value = obj.email;
      form.inputData.phone.value = (obj.phoneNumber === "0000000000") || (obj.phoneNumber === "(000) 000-0000") ? "" : obj.phoneNumber;

      Object.keys(form.inputData).forEach(key => {
        if (key !== "emailId") {
          const item = form.inputData[key];
          item.disabled = true;
          item.error = "";
        }
      });

      this.setState({form});
      this.props.dispatchDiscardChanges(false);
    }
  }

  bubbleValue (evt, key, error) {
    const targetVal = evt.target.value;
    this.setState(state => {
      state = {...state};
      state.form.inputData[key].value = targetVal;
      state.form.inputData[key].error = error;
      return state;
    });
  }

  onChange (evt, key) {
    if (evt && evt.target) {
      if (evt.target.pattern && evt.target.checkValidity()) {
        this.invalid[key] = false;
      }
      const targetVal = evt.target.value;
      this.setState(state => {
        state = {...state};
        const inputData = state.form.inputData;
        state.form.inputData[key].value = targetVal;
        inputData[key].error = !this.invalid[key] ? "" : inputData[key].error;
        this.invalid[key] = false;
        return {
          ...state
        };
      });
    }
  }

  displayManageProfileNotification () {
    let template = this.props.modalsMeta && this.props.modalsMeta.PASSWORD_RESET_SELLER;
    template = {templateName: "StatusModalTemplate", image: "", ...template};
    this.props.toggleModal(TOGGLE_ACTIONS.SHOW, {...template});
    mixpanel.trackEvent(MIXPANEL_CONSTANTS.USER_PROFILE.CHANGE_PASSWORD.DISPLAY_SELLER_MANAGE_PROFILE, {WORK_FLOW: "EDIT_USER_PROFILE"});
  }

  displayChangePassword() {
    const meta = { templateName: "ResetPasswordTemplate" };
    this.props.toggleModal(TOGGLE_ACTIONS.SHOW, {...meta});
    mixpanel.trackEvent(MIXPANEL_CONSTANTS.USER_PROFILE.CHANGE_PASSWORD.DISPLAY_CHANGE_PASSWORD, {WORK_FLOW: "EDIT_USER_PROFILE"});
  }

  disableInput (disable) {
    disable = !!disable;
    if (disable && this.isDirty()) {
      const meta = { templateName: "Alert" };
      this.props.toggleModal(TOGGLE_ACTIONS.SHOW, {...meta});
      mixpanel.trackEvent(MIXPANEL_CONSTANTS.USER_PROFILE.EDIT_USER_PROFILE.CANCEL_EDIT_PROFILE, {WORK_FLOW: "EDIT_USER_PROFILE"});
    } else {
      const form = {...this.state.form};
      form.isDisabled = disable;
      form.inputData.phone.disabled = disable;
      Object.keys(form.inputData).forEach(key => {
        if (key !== "emailId") {
          const item = form.inputData[key];
          item.disabled = disable;
          item.error = "";
        }
      });
      this.setState({form});
      if (!disable) mixpanel.trackEvent(MIXPANEL_CONSTANTS.USER_PROFILE.EDIT_USER_PROFILE.EDIT_PROFILE, {WORK_FLOW: "EDIT_USER_PROFILE"});
    }
  }

  // toggleUnderwritingCheck () {
  //   const form = {...this.state.form};
  //   form.underwritingChecked = !form.underwritingChecked;
  //   this.setState({form});
  // }

  async saveUser (evt) {
    evt.preventDefault();

    if (!this.validateState()) {
      this.loader("form", true);
      const loginId = this.state.form.inputData.emailId.value;
      const firstName = this.state.form.inputData.firstName.value;
      const lastName = this.state.form.inputData.lastName.value;
      const phoneNumber = this.state.form.inputData.phone.value ? this.state.form.inputData.phone.value : "0000000000"; //[note:to handle VIP phone number validation]
      const payload = {
        user: {
          loginId,
          firstName,
          lastName,
          phoneNumber,
          properties: {
            companyName: this.state.form.inputData.companyName.value
          }
        }
      };

      const url = this.state.form.apiPath;
      if (this.isDirty()) {
        const mixpanelPayload = {
          API: url,
          WORK_FLOW: "EDIT_USER_PROFILE"
        };
        return Http.put(`${url}/${payload.user.loginId}`, payload, null, null, this.props.showNotification, this.state.form.profileSaveMessage)
          .then(async res => {
            this.loader("form", false);
            this.props.updateUserProfile(res.body);
            this.disableInput(true);
            mixpanelPayload.API_SUCCESS = true;
          })
          .catch(err => {
            this.loader("form", false);
            mixpanelPayload.API_SUCCESS = false;
            mixpanelPayload.ERROR = err.message ? err.message : err;
          })
          .finally(() => {
            mixpanel.trackEvent(MIXPANEL_CONSTANTS.USER_PROFILE.EDIT_USER_PROFILE.SAVE_PROFILE, mixpanelPayload);
          });
      } else {
        this.loader("form", false);
        this.disableInput(true);
      }
    }
    return null;
  }

  render () {
    return (
      <div className={`user-profile-content h-100${this.state.form.loader ? " loader" : ""}`}>
        <header className={this.state.section.headerClasses}>{this.state.section.sectionTitle}</header>
        <form className={this.state.section.formClasses} autoComplete="off" onSubmit={this.saveUser} style={{marginRight: "50%"}}>
          {this.getFieldRenders()}
        </form>
      </div>
    );
  }
}

UserProfile.propTypes = {
  dispatchDiscardChanges: PropTypes.func,
  shouldDiscard: PropTypes.bool,
  showNotification: PropTypes.func,
  toggleModal: PropTypes.func,
  userProfile: PropTypes.object,
  updateUserProfile: PropTypes.func
};

const mapStateToProps = state => {
  return {
    modalsMeta: state.content.metadata ? state.content.metadata.MODALSCONFIG : {},
    userProfileContent: state.content && state.content.metadata && state.content.metadata.SECTIONSCONFIG && state.content.metadata.SECTIONSCONFIG.USERPROFILE,
    shouldDiscard: state.modal.shouldDiscard,
    userProfile: state.user.profile
  };
};

const mapDispatchToProps = {
  dispatchDiscardChanges,
  showNotification,
  toggleModal,
  updateUserProfile
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(UserProfile);
