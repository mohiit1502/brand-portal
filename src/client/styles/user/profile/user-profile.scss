@import "../../override/variables";

.user-profile-content {
  .password-reset-row {
    .badge {
      border: 1px solid $blue;
    }
  }
}
