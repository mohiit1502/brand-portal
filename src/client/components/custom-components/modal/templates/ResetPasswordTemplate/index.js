import ResetPasswordTemplate from "./ResetPasswordTemplate.jsx";

export default ResetPasswordTemplate;
