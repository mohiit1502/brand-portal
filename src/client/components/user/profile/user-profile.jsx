import React from "react";
import { connect } from "react-redux";
import "../../../styles/user/profile/user-profile.scss";
import PropTypes from "prop-types";
import CustomInput from "../../custom-input/custom-input";

class UserProfile extends React.Component {

  constructor (props) {
    super(props);
    this.onInputChange = this.onInputChange.bind(this);
    this.setFormData = this.setFormData.bind(this);
    this.disableInput = this.disableInput.bind(this);
    this.toggleUnderwritingCheck = this.toggleUnderwritingCheck.bind(this);

    this.state = {
      form: {
        isDisabled: true,
        underwritingChecked: false,
        id: "user-profile-form",
        inputData: {
          firstName: {
            label: "First Name",
            required: true,
            value: "",
            type: "text",
            pattern: null,
            disabled: true
          },
          lastName: {
            label: "Last Name",
            required: true,
            value: "",
            type: "text",
            pattern: null,
            disabled: true
          },
          companyName: {
            label: "Company Name",
            required: true,
            value: "",
            type: "text",
            pattern: null,
            disabled: true
          },
          emailId: {
            label: "Email ID",
            required: true,
            value: "",
            type: "email",
            pattern: null,
            disabled: true
          },
          phone: {
            label: "Mobile Number",
            required: false,
            value: "",
            type: "text",
            pattern: null,
            disabled: true
          }
        }
      }
    };
  }

  componentDidMount() {
    const obj = {
      firstName: "Shubhansh",
      lastName: "Sahai",
      companyName: "Walmart Labs",
      emailId: "shubhansh.sahai@walmartlabs.com",
      phone: "9686648597"
    };

    this.setFormData(obj);

  }

  setFormData(obj) {
    const form = {...this.state.form};
    form.inputData.firstName.value = obj.firstName;
    form.inputData.firstName.disabled = true;
    form.inputData.lastName.value = obj.lastName;
    form.inputData.lastName.disabled = true;
    form.inputData.companyName.value = obj.companyName;
    form.inputData.companyName.disabled = true;
    form.inputData.emailId.value = obj.emailId;
    form.inputData.emailId.disabled = true;
    form.inputData.phone.value = obj.phone;
    form.inputData.phone.disabled = true;

    this.setState({
      form
    });
  }

  onInputChange (evt, key) {
    if (evt && evt.target) {
      const targetVal = evt.target.value;
      this.setState(state => {
        state = {...state};
        state.form.inputData[key].value = targetVal;
        return {
          ...state
        };
      });
    }
  }

  disableInput (disable) {
    disable = !!disable;

    const form = {...this.state.form};

    form.isDisabled = disable;

    form.inputData.firstName.disabled = disable;
    form.inputData.lastName.disabled = disable;
    form.inputData.companyName.disabled = disable;
    form.inputData.emailId.disabled = disable;
    form.inputData.phone.disabled = disable;

    this.setState({
      form
    });

  }

  toggleUnderwritingCheck () {
    const form = {...this.state.form};
    form.underwritingChecked = !form.underwritingChecked;
    this.setState({form});
  }

  render () {

    return (
      <div className="row user-profile-content h-100">
        <div className="col h-100">
          <div className="row h-10">
            <div className="col">
              <h3>User Profile</h3>
            </div>
          </div>
          <div className="row mt-4 h-90">
            <div className="col">
              <form className="h-100" autoComplete="off">
                <div className="row h-60">
                  <div className="col h-100">
                    <div className="row">
                      <div className="col-xl-3 col-5">
                        <CustomInput key={"firstName"}
                          inputId={"firstName"}
                          formId={this.state.form.id} label={this.state.form.inputData.firstName.label}
                          required={this.state.form.inputData.firstName.required} value={this.state.form.inputData.firstName.value}
                          type={this.state.form.inputData.firstName.type} pattern={this.state.form.inputData.firstName.pattern}
                          onChangeEvent={this.onInputChange} disabled={this.state.form.inputData.firstName.disabled} />
                      </div>
                      <div className="col-xl-3 col-5">
                        <CustomInput key={"lastName"}
                          inputId={"lastName"}
                          formId={this.state.form.id} label={this.state.form.inputData.lastName.label}
                          required={this.state.form.inputData.lastName.required} value={this.state.form.inputData.lastName.value}
                          type={this.state.form.inputData.lastName.type} pattern={this.state.form.inputData.lastName.pattern}
                          onChangeEvent={this.onInputChange} disabled={this.state.form.inputData.lastName.disabled} />
                      </div>
                    </div>
                    <div className="row">
                      <div className="col-xl-3 col-5">
                        <CustomInput key={"companyName"}
                          inputId={"companyName"}
                          formId={this.state.form.id} label={this.state.form.inputData.companyName.label}
                          required={this.state.form.inputData.companyName.required} value={this.state.form.inputData.companyName.value}
                          type={this.state.form.inputData.companyName.type} pattern={this.state.form.inputData.companyName.pattern}
                          onChangeEvent={this.onInputChange} disabled={this.state.form.inputData.companyName.disabled} />
                      </div>
                    </div>
                    <div className="row">
                      <div className="col-xl-3 col-5">
                        <CustomInput key={"emailId"}
                          inputId={"emailId"}
                          formId={this.state.form.id} label={this.state.form.inputData.emailId.label}
                          required={this.state.form.inputData.emailId.required} value={this.state.form.inputData.emailId.value}
                          type={this.state.form.inputData.emailId.type} pattern={this.state.form.inputData.emailId.pattern}
                          onChangeEvent={this.onInputChange} disabled={this.state.form.inputData.emailId.disabled} />
                      </div>
                      <div className="col-xl-3 col-5">
                        <CustomInput key={"phone"}
                          inputId={"phone"}
                          formId={this.state.form.id} label={this.state.form.inputData.phone.label}
                          required={this.state.form.inputData.phone.required} value={this.state.form.inputData.phone.value}
                          type={this.state.form.inputData.phone.type} pattern={this.state.form.inputData.phone.pattern}
                          onChangeEvent={this.onInputChange} disabled={this.state.form.inputData.phone.disabled} />
                      </div>
                    </div>
                    <div className="row mb-5 password-reset-row">
                      <div className="col-3">
                        <div className="btn btn-primary btn-sm no-border-radius px-3">
                          Password Reset
                        </div>
                        <span className="badge badge-pill bg-white text-blue ml-2 cursor-pointer"> ? </span>
                      </div>
                    </div>
                  </div>
                </div>
                <div className="row h-40">
                  <div className="col h-100">
                    <div className="row pt-5">
                      <div className="col-xl-6 col-10">
                        <label>
                          <input type="checkbox" className="mr-2" checked={this.state.form.underwritingChecked} onChange={this.toggleUnderwritingCheck}/>
                          I have a good faith belief that the use of the material in the manner complained of is not authorized by the copyright owner, its agent, or the law.
                        </label>
                      </div>
                    </div>
                    <div className="row pt-5">
                      <div className="col-xl-3 col-5" />
                      <div className="col-xl-3 col-5 text-right">
                        {
                          this.state.form.isDisabled ?
                            (
                              <div className="btn btn-primary btn-sm no-border-radius px-4" onClick={() => this.disableInput(false)}>
                                Edit
                              </div>
                            ) :
                            (
                              <span>
                                <div className="btn btn-link font-size-14 px-4 mr-3" onClick={() => this.disableInput(true)}>Cancel</div>
                                <button type="submit" className="btn btn-primary btn-sm no-border-radius px-4 font-size-14" disabled={!this.state.form.underwritingChecked}>
                                  Save
                                </button>
                              </span>
                            )
                        }

                      </div>
                    </div>
                  </div>
                </div>

              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => state;

export default connect(
  mapStateToProps,
  dispatch => ({dispatch})
)(UserProfile);
