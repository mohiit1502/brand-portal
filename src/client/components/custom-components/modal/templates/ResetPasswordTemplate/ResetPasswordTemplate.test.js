/* eslint-disable filenames/match-regex, no-unused-vars, no-undef */
import React from "react";
import ResetPasswordTemplate from "./ResetPasswordTemplate";
import {Provider} from "react-redux";
import "core-js/stable";
import "regenerator-runtime/runtime";
import MODALSMETA from "../../../../../config/contentDescriptors/modals-meta";
import FORMFIELDMETA from "../../../../../../../src/client/config/formsConfig/form-field-meta";
import profile from "../../../../../../../test/client/mocks/userProfile";
import renderer, {act} from "react-test-renderer";
import Cookies from "electrode-cookies";
import {
  mockFailFetch,
  mockFetch,
  testStore,
  update
} from "../../../../../utility/TestingUtils";
import {mount} from "enzyme";

let store;
jest.mock("electrode-cookies");

const setUpRender = () => {
  const mockStore = {
    content: {
      metadata: {...MODALSMETA, ...FORMFIELDMETA},
    },
    user: {profile},
    modal:{shouldDiscard : false}
  };
  store = testStore(mockStore);
  return renderer.create(<Provider store={store}><ResetPasswordTemplate /></Provider>);
};

const setUpMount = () => {
  const mockStore = {
    content: {
      metadata: {...MODALSMETA, ...FORMFIELDMETA},
    },
    user: {profile},
    modal:{shouldDiscard : false}
  };
  store = testStore(mockStore);
  return mount(<Provider store={store}><ResetPasswordTemplate /></Provider>);
};

describe("ResetPasswordTemplate", () => {

  let wrapper;
  beforeEach(() => {
    jest.restoreAllMocks();
    jest.resetAllMocks();
    Cookies.get.mockImplementation(jest.fn());
  });

   it("renders without error", () => {
     wrapper = setUpRender();
     expect(wrapper.toJSON()).toMatchSnapshot();
   });

   it("trigger event", () => {
     wrapper = setUpMount();
     wrapper.find(".close").at(0).simulate("click");
     update(wrapper)
   });
});
