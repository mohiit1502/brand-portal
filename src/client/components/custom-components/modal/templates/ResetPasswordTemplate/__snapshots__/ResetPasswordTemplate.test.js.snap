// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPasswordTemplate renders without error 1`] = `
<div
  className="c-ResetPasswordTemplate modal fade show"
  id="singletonModal"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog modal-dialog-centered"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header align-items-center"
      >
        Change Password
        <button
          aria-label="Close"
          className="close text-white"
          onClick={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
      <div
        className="modal-body text-left p-0"
      >
        <form
          className="h-100 px-4"
          onSubmit={[Function]}
        >
          <p
            className="pb-1 pt-4"
          >
            Please type your current password and then create the new one.
          </p>
          
          
          <div
            className="px-0 col-12"
          >
            <div
              className="form-group custom-input-form-group form-group-text mb-3
        "
              style={
                Object {
                  "position": "static",
                }
              }
            >
              <input
                className="form-control form-control-currentPassword custom-input-element"
                id="resetPassword-currentPassword-custom-input"
                onBlur={[Function]}
                onChange={[Function]}
                onInvalid={[Function]}
                required={false}
                type="password"
                value=""
              />
              
              <label
                className="custom-input-label custom-input-label-placeholder"
                htmlFor="resetPassword-currentPassword-custom-input"
              >
                <div
                  className="label-upper-bg position-absolute w-100 h-50 d-block"
                />
                <div
                  className="label-lower-bg position-absolute w-100 h-50 d-block"
                />
                <span
                  className="label-text required"
                >
                   
                  Current Password
                   
                </span>
              </label>
              <small
                className="form-text custom-input-help-text text-muted"
                style={
                  Object {
                    "paddingLeft": undefined,
                  }
                }
              >
                
              </small>
            </div>
          </div>
          <div
            className="px-0 col-12"
          >
            <div
              className="form-group custom-input-form-group form-group-text mb-3
        "
              style={
                Object {
                  "position": "static",
                }
              }
            >
              <input
                className="form-control form-control-newPassword custom-input-element"
                id="resetPassword-newPassword-custom-input"
                onBlur={[Function]}
                onChange={[Function]}
                onInvalid={[Function]}
                required={false}
                type="password"
                value=""
              />
              
              <label
                className="custom-input-label custom-input-label-placeholder"
                htmlFor="resetPassword-newPassword-custom-input"
              >
                <div
                  className="label-upper-bg position-absolute w-100 h-50 d-block"
                />
                <div
                  className="label-lower-bg position-absolute w-100 h-50 d-block"
                />
                <span
                  className="label-text required"
                >
                   
                  New Password
                   
                </span>
              </label>
              <small
                className="form-text custom-input-help-text text-muted"
                style={
                  Object {
                    "paddingLeft": undefined,
                  }
                }
              >
                
              </small>
            </div>
          </div>
          <div
            className="px-0 col-12"
          >
            <div
              className="form-group custom-input-form-group form-group-text mb-2
        "
              style={
                Object {
                  "position": "static",
                }
              }
            >
              <input
                className="form-control form-control-confirmNewPassword custom-input-element"
                id="resetPassword-confirmNewPassword-custom-input"
                onBlur={[Function]}
                onChange={[Function]}
                onInvalid={[Function]}
                required={false}
                type="password"
                value=""
              />
              
              <label
                className="custom-input-label custom-input-label-placeholder"
                htmlFor="resetPassword-confirmNewPassword-custom-input"
              >
                <div
                  className="label-upper-bg position-absolute w-100 h-50 d-block"
                />
                <div
                  className="label-lower-bg position-absolute w-100 h-50 d-block"
                />
                <span
                  className="label-text required"
                >
                   
                  Confirm New Password
                   
                </span>
              </label>
              <small
                className="form-text custom-input-help-text text-muted"
                style={
                  Object {
                    "paddingLeft": undefined,
                  }
                }
              >
                
              </small>
            </div>
          </div>
          <div
            className=" col-12"
          />
          <div
            className="c-ButtonsPanel form-row pr-4 text-right bg-blue py-3 mx-n4"
            data-test="button-panel"
          >
            <div
              className="col reset-password-button-panel text-right pr-0"
              data-test="button-div"
            >
              <button
                className="btn btn-sm cancel-btn text-primary"
                onClick={[Function]}
                type="button"
              >
                Back
              </button>
              <button
                className="btn btn-sm btn-primary submit-btn px-3 ml-3"
                type="submit"
              >
                Change Password
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
